lib/utils/auth_utils.dart

```diff
@@ -37,7 +37,13 @@ class AuthUtils {
         return;
       }
       await context.read<AuthCubit>().refreshSession();
-      navigate();
+
+      /// Because the session is refreshed and the router needs to respond
+      /// we will delay the navigation by 100 milliseconds.
+      Future.delayed(
```

Comment:
This doesnt sound like a really good solution. It probably works but we don't know how long exactly we need to wait. It might even be different per phone (slow/older phones could be slower?)

Reply:
Is there a way to do this differently?

Reply:
Nope, it's not ideal. I am thinking of a different way to do it. As this is cause from the routing as we have the refresh part

Reply:
Should be fine now